### PR TITLE
Support for TDBID

### PR DESCRIPTION
**Pull request by Roms1383**
It would be great to add support for more types :)
Please note that I haven't implemented table lookup yet.

### CODE REVIEW

File: red4ext-sys/src/interop.rs [hunk 1]
```diff
@@ -60,7 +61,7 @@ impl TweakDBID {
     #[inline]
     pub const fn new(str: &str) -> Self {
         assert!(str.len() <= u8::MAX as usize);
-        let hash = crc32(str.as_bytes(), 0).to_ne_bytes() as [u8; 4];
+        let hash = crc32(str.as_bytes()).to_ne_bytes() as [u8; 4];
```
Comment (jac3km4) [inline red4ext-sys/src/interop.rs:64]: I think the `as ...` but is unnecessary

File: red4ext-sys/src/interop.rs [hunk 2]
```diff
@@ -71,7 +72,7 @@ impl TweakDBID {
         let bytes = base.hash.to_ne_bytes() as [u8; 8];
         assert!((bytes[4] as usize + str.len()) <= u8::MAX as usize);
         let seed = u32::from_ne_bytes([bytes[0], bytes[1], bytes[2], bytes[3]]);
-        let hash = crc32(str.as_bytes(), seed).to_ne_bytes() as [u8; 4];
+        let hash = crc32_seed(str.as_bytes(), seed).to_ne_bytes() as [u8; 4];
```
Comment (jac3km4) [inline red4ext-sys/src/interop.rs:75]: same as above

File: .gitignore [hunk 3]
```diff
@@ -1 +1,2 @@
 /target
+/deps
```
Comment (jac3km4) [inline .gitignore:2]: Is this needed?

Comment (Roms1383) [reply .gitignore:2]: there's always a `deps` folder that pops up, not sure from which tool : C++ maybe ?

Comment (Roms1383) [reply .gitignore:2]: otherwise let me just remove it ...

File: red4ext-sys/src/interop.rs [hunk 4]
```diff
@@ -44,6 +45,54 @@ unsafe impl ExternType for CName {
...
+}
+
+impl TweakDBID {
+    #[inline]
+    pub const fn new(str: &str) -> Self {
+        assert!(str.len() <= u8::MAX as usize);
+        let hash = crc32(str.as_bytes()).to_ne_bytes();
+        let length = str.len() as u8;
+        Self {
+            hash: u64::from_ne_bytes([hash[0], hash[1], hash[2], hash[3], length, 0u8, 0u8, 0u8]),
```
Comment (jac3km4) [inline red4ext-sys/src/interop.rs:67]: I'm wondering if it wouldn't be easier to do it the other way around,
represent the struct as `crc: u32, len: u8, padding: [u8; 3]`

Comment (Roms1383) [reply red4ext-sys/src/interop.rs:67]: that's what I did originally in the commits prior to [cd15696](https://github.com/jac3km4/red4ext-rs/pull/5/commits/cd15696966b6b26ebbf307fc90c134558f55fa45)

**Pull request merged**